### DISCUSSION TITLE
Login Using password

### DISCUSSION THREAD

**Discussion by KetulCodiste** [Web3Auth/Web3Auth#480] [Help]
As much as I know, web3Auth provides the login process in the form of social platforms like facebook, discord, google, twitter, etc and through the magic link that web3Auth provides in the mail to login. But is there any way to sign up and login using the password. Like the normal flow that all the websites provides, enter the username and password for the authentication. Is it possible to achieve this kind of behaviour using web3Auth?

**Comment by himanshuchawla009** [comment]
You can implement your own, email and password server or use some service. The only thing that web3auth depends on is that OAuth 2.0 id_token compatible jwt token should be issued by the email/password service and then you can feed that token to web3auth sdk.

So steps to implement email password /custom auth will look like this:-

- Implement your own email/password auth server or use some third-party service.
- Generate a jwt id_token from that service on user login.
- Expose a JWKS Endpoint from service.
- Create a custom verifier with your JWKS Endpoint and JWT token details on https://dashboard.web3auth.io
- Use this verifier and feed your jwt token in web3auth sdk. 

A similar example for custom auth can be found here:-  https://github.com/Web3Auth/Web3Auth/tree/master/demo/firebase-react-app

**Reply by shahbaz17** [reply]
Also, take a look at https://github.com/orgs/Web3Auth/discussions/476 on how to pass the JWT token to Web3Auth SDK.

**Reply by KetulCodiste** [reply]
now, I am using the auth0, 3rd party service provider for authentication. I have followed all the steps required for the authentication purpose. But while logging in, I am getting the following error:

Could not get result from torus nodes Error occurred while verifying paramscould not verify identity my_email auth0 | my_auth0_clientId

**Reply by himanshuchawla009** [reply]
checkout this doc to troubleshoot the error: https://web3auth.io/docs/troubleshooting/jwt-errors#error-occurred-while-verifying-params-could-not-verify-identity
